backend/ccm/canvas_api/course_api_handler.py

```diff
@@ -0,0 +1,93 @@
+import logging
+from http import HTTPStatus
+from rest_framework.views import APIView
+from rest_framework import authentication, permissions
+from rest_framework.response import Response
+from rest_framework.request import Request
+
+from canvasapi.exceptions import CanvasException
+from canvasapi.course import Course
+from canvasapi import Canvas
+
+from backend.ccm.canvas_api.canvasapi_serializer import CourseSerializer
+from .exceptions import CanvasHTTPError
+from canvas_oauth.exceptions import InvalidOAuthReturnError
+
+from backend.ccm.canvas_api.canvas_credential_manager import CanvasCredentialManager
+
+from drf_spectacular.utils import extend_schema
+
+from rest_framework_tracking.mixins import LoggingMixin
+
+logger = logging.getLogger(__name__)
+
+CANVAS_CREDENTIALS = CanvasCredentialManager()
+
+class CanvasCourseAPIHandler(LoggingMixin, APIView):
+
+    logging_methods = ['GET', 'PUT']
+    """
+    API handler for Canvas course data.
+    """
+    authentication_classes = [authentication.SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get(self, request: Request, course_id: int) -> Response:
+        """
+        Get course data from Canvas.
+        """
+        try:
+            canvas_api: Canvas = CANVAS_CREDENTIALS.get_canvasapi_instance(request)
+            logger.info(f"Getting course data for course_id: {course_id}")
+            
+            # Call the Canvas API package to get course details.
+            course: Course = canvas_api.get_course(course_id)
+            logger.info(f"Course data retrieved: {course}")
+            
+            # Format the course object to return specific course info
+            formatted_course = {
+                "id": course.id,
+                "name": course.name,
+                "enrollment_term_id": course.enrollment_term_id
+            }
+            
+            return Response(formatted_course, status=HTTPStatus.OK)
+        except CanvasException as e:
```

Comment:
I feel like these 2 errors are redundant between these methods and probably will be repeated in multiple places. We probably should have a centralized method like `handle_canvas_error`? It seems like you have one like that in canvas_credential_manager, just maybe need to move that to some other file? I'm not sure but I feel like there will just be a lot of code duplication here as this keeps being developed.

Reply:
I have refactored the code with a simplified error approach 
https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/503/commits/3baba4701542b2a02a769721c76ba3766d19ef89